### PR TITLE
applications: serial_lte_modem: Fix to PGPS, did not work without pre…

### PR DESCRIPTION
**Pull request by VTPeltoketo**
…vious fix

- pgps_event_handler changed, correct error handling after read_agps_req call
- GNSS documentation changed, XGPSDEL is only a debugging feature
- GNSS documentation changed, correct range of interval parameter

### CODE REVIEW

**Comment by VTPeltoketo** [comment]
> oh, I thought it was 79....

**Review by junqingzou — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

File: applications/serial_lte_modem/doc/GNSS_AT_commands.rst [hunk 1]
```diff
@@ -682,11 +682,13 @@ Delete GNSS data
 
 The ``#XGPSDEL`` command deletes GNSS data from non-volatile memory.
 This command should be issued when GNSS is activated but not started yet.
+Note! This is considered a debug feature, and is not supposed to be used in production code.
```
Comment (MarkusLassila) [inline applications/serial_lte_modem/doc/GNSS_AT_commands.rst:685]: Should this use note:: -syntax?

File: applications/serial_lte_modem/doc/GNSS_AT_commands.rst [hunk 2]
```diff
@@ -682,11 +682,13 @@ Delete GNSS data
...
 
 Set command
 -----------
 
 The set command allows you to delete old GNSS data.
+The usage of the command does not trigger A-GPS request event and thus may delay the full functionality of A-GPS and P-GPS until the next periodic A-GPS request has been gotten.
```
Comment (MarkusLassila) [inline applications/serial_lte_modem/doc/GNSS_AT_commands.rst:691]: Can you split this to multiple sentences?

File: applications/serial_lte_modem/doc/GNSS_AT_commands.rst (hunk 1 above)
Comment (VTPeltoketo) [reply applications/serial_lte_modem/doc/GNSS_AT_commands.rst]: Fixed

File: applications/serial_lte_modem/doc/GNSS_AT_commands.rst (hunk 2 above)
Comment (VTPeltoketo) [reply applications/serial_lte_modem/doc/GNSS_AT_commands.rst]: Fixed

**Review by MarkusLassila — APPROVED** [review]

File: applications/serial_lte_modem/doc/GNSS_AT_commands.rst [hunk 3]
```diff
@@ -683,10 +683,14 @@ Delete GNSS data
...
 This command should be issued when GNSS is activated but not started yet.
 
+.. note::
+   This is considered a debug feature, and is not supposed to be used in production code.
+
 Set command
 -----------
 
 The set command allows you to delete old GNSS data.
+The usage of the command does not trigger A-GPS request event. The execution of the command may delay the full functionality of A-GPS and P-GPS until the next periodic A-GPS request has been received.
```
Comment (peknis) [inline applications/serial_lte_modem/doc/GNSS_AT_commands.rst:693]:
```suggestion
Using this command does not trigger A-GPS request event.
The execution of the command may delay the full functionality of A-GPS and P-GPS until the next periodic A-GPS request has been received.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Pull request merged**